packages/framework/client-logger/external-telemetry/src/container/telemetryProducer.ts

```diff
@@ -63,14 +60,14 @@ export class ContainerEventTelemetryProducer {
 		return telemetry;
 	}
 
-	private produceBasicContainerTelemetry = <T extends IContainerTelemetry>(
+	private produceBasicContainerTelemetry = <IContainerTelemetry>(
 		eventName: ContainerTelemetryEventName,
-	): T => {
+	): IContainerTelemetry => {
 		return {
 			eventName,
 			containerId: this.getContainerId(),
 			documentId: this.getDocumentId(),
-		} as T;
+		} as IContainerTelemetry;
```

Comment:
The object naturally matches the interface, the cast should not be necessary.
```suggestion
		};
```

Reply:
Updated.